tests/test_array.py

```diff
@@ -465,6 +466,28 @@ def test_transpose(self):
 
         self.assertRaises(Exception, darray.transpose, 'invalid')
 
+    def test_random(self):
```

Comment:
to test that two arrays are completely equivalent (including shape, sparsity, etc...) you can use the equal(arr1, arr2) function which checks everything, maybe it's useful here